backend/src/main/java/eventlistener/controller/exception/ControllerAdvisor.java

```diff
@@ -0,0 +1,62 @@
+package eventlistener.controller.exception;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.security.SignatureException;
+import java.util.NoSuchElementException;
+
+@Slf4j
+@ControllerAdvice
+public class ControllerAdvisor extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(NoSuchElementException.class)
+    public ResponseEntity<ApiError> handleNoSuchElementException(NoSuchElementException ex){
+        log.error("Resource not found!",ex);
```

Comment:
You should decided if you want to put a "!" after your exception message or not :-)